src/core/auth/google_verifier.go

```diff
@@ -0,0 +1,86 @@
+package google_verifer
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	grpc_auth "github.com/grpc-ecosystem/go-grpc-middleware/auth"
+	"github.com/grpc-ecosystem/go-grpc-middleware/util/metautils"
+	"io/ioutil"
+	"log"
+	"net/http"
+)
+
+type Google_verifier struct {
+	Url string
+}
+
+func New(url string) Google_verifier {
```

Comment:
You should probably return a pointer here, as the `Google_verifier` struct could technically grow in size.